### PR TITLE
[FLINK-13434][e2e] Add stop-with-savepoint end-to-end tests.

### PR DESCRIPTION
**Pull request by kl0u**
## What is the purpose of the change

The purpose of this PR is to add some end-to-end tests for the newly introduced `stop-with-savepoint` command. 

Instead of creating a new test (in bash), this PR modifies already existing tests to use the new feature. In the future, we should include an end-to-end test with a more complicated job that does some state manipulation at the `notifyCheckpointComplete`, as the exactly-once sinks do. This will *actually* test the new feature.

## Brief change log

This PR modifies the `test_stateful_stream_job_upgrad` and the `test_resume_savepoint`, without changing their logic.

## Verifying this change

This change is already covered by existing tests, which are the ones modified.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by kl0u** [comment]
R @azagrebin

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 1]
```diff
@@ -471,6 +471,11 @@ function wait_job_terminal_state {
   done
 }
 
+function stop_with_savepoint {
+  echo "INSIDE: Stopping job $1 with a savepoint @ $2."
```
Comment (aljoscha) [inline flink-end-to-end-tests/test-scripts/common.sh]: Is the printing meant to stay here?

Comment (kl0u) [reply flink-end-to-end-tests/test-scripts/common.sh]: Nope, I will remove it. It stayed after debugging.

**Comment by kl0u** [comment]
Merged.

**Pull request closed without merging**